rllib/models/action_dist_v2.py

```diff
@@ -0,0 +1,119 @@
+        >>> model = ... # a model that outputs a vector of logits
+        >>> action_logits = model.forward(obs)
+        >>> action_dist = ActionDistribution(action_logits)
+        >>> action = action_dist.sample()
+        >>> logp = action_dist.logp(action)
+        >>> kl = action_dist.kl(action_dist2)
+        >>> entropy = action_dist.entropy()
+
+    """
+
+    @abc.abstractmethod
+    def sample(
+        self,
+        *,
+        sample_shape: Tuple[int, ...] = None,
+        return_logp: bool = False,
+        **kwargs
+    ) -> Union[TensorType, Tuple[TensorType, TensorType]]:
+        """Draw a sample from the action distribution.
+
+        Args:
+            sample_shape: The shape of the sample to draw.
+            return_logp: Whether to return the logp of the sampled action.
+            **kwargs: Forward compatibility placeholder.
+
+        Returns:
+            The sampled action. If return_logp is True, returns a tuple of the
+            sampled action and its logp.
+        """
+
+    @abc.abstractmethod
+    def rsample(
+        self,
+        *,
+        sample_shape: Tuple[int, ...] = None,
+        return_logp: bool = False,
+        **kwargs
+    ) -> Union[TensorType, Tuple[TensorType, TensorType]]:
+        """Draw a re-parameterized sample from the action distribution.
+
+        Args:
+            sample_shape: The shape of the sample to draw.
+            return_logp: Whether to return the logp of the sampled action.
+            **kwargs: Forward compatibility placeholder.
+
+        Returns:
+            The sampled action. If return_logp is True, returns a tuple of the
+            sampled action and its logp.
+        """
+
+    @abc.abstractmethod
+    def logp(self, action: TensorType, **kwargs) -> TensorType:
+        """The log-likelihood of the action distribution.
+
+        Args:
+            action: The action to compute the log-likelihood for.
+            **kwargs: Forward compatibility placeholder.
+
+        Returns:
+            The log-likelihood of the action.
+        """
+
+    @abc.abstractmethod
+    def kl(self, other: "ActionDistributionV2", **kwargs) -> TensorType:
+        """The KL-divergence between two action distributions.
+
+        Args:
+            other: The other action distribution.
+            **kwargs: Forward compatibility placeholder.
+
+        Returns:
+            The KL-divergence between the two action distributions.
+        """
+
+    @abc.abstractmethod
+    def entropy(self, **kwargs) -> TensorType:
+        """The entropy of the action distribution.
+
+        Args:
+            **kwargs: Forward compatibility placeholder.
+
+        Returns:
+            The entropy of the action distribution.
+        """
+
+    @staticmethod
+    @abc.abstractmethod
+    def required_model_output_shape(
+        action_space: gym.Space, model_config: ModelConfigDict
+    ) -> Tuple[int, ...]:
+        """Returns the required shape of an input parameter tensor for a
```

Comment:
Can we give 2 examples here of how this method will be used (for Categorical and DiagGaussian)?
* Does the shape include the batch dim (time dim, etc..)?
* What are typical cases, where the model config would be used and how would it be used?

Reply:
I am not sure if this will actually get used tbh. I have it here just as a reminder to the next developer that this was part of the old distribution that we may or may not keep depending on how the catalog gets written.  added a comment to reflect this.